### ISSUE TITLE
Rename StartLimitInterval to StartLimitIntervalSec for all systemd services

### ISSUE THREAD

**Issue by nh2** [NixOS/nixpkgs#45785]
See https://github.com/systemd/systemd/issues/4826#issuecomment-264890288:

> See https://www.freedesktop.org/software/systemd/man/systemd.unit.html#StartLimitIntervalSec=.
(Note: the setting was renamed to have the "Sec" suffix in systemd-230.)

There are 20 remaining occurrences of the old `StartLimitInterval` in nixpkgs services:

```
git grep 'StartLimitIntervalSec' 74495e4f3afe5f5467dce80842105e9b841a8c91 | wc -l
1

git grep 'StartLimitInterval' 74495e4f3afe5f5467dce80842105e9b841a8c91 | wc -l
21
```

---

**Update** from https://github.com/NixOS/nixpkgs/issues/45786#issuecomment-440088882:

* [ ] We should probably also try and make usage of `StartLimitInterval` in the old section emit a warning when it's used, as having it in the new section has [benefits](https://github.com/NixOS/nixpkgs/issues/45786#issuecomment-440091879)

**Comment by vcunat** [comment]
Oh, yes, that's their "great" policy of ignoring unknown options...

**Comment by nh2** [comment]
@vcunat `StartLimitInterval` doens't get ignored, it has a backwards-compatible `/* obsolete alias */` in the code:

https://github.com/systemd/systemd/blob/2af83a2e41e0eaf6bf434b2afb652006833879d5/src/core/main.c#L673

But the stuff in #45786 is ignoring unknown options.

**Comment by vcunat** [comment]
Oh, I didn't notice the alias (it's not in their docs).  Systemd generally ignores unknown options (silently IIRC) &ndash; that's what I meant.

**Comment by nh2** [comment]
I've updated the issue description to include https://github.com/NixOS/nixpkgs/issues/45786#issuecomment-440091879

**Comment by nh2** [comment]
I noticed another detail:

`serviceConfig.StartLimitIntervalSec` is especially annoying because even though it is now undocumented and deprecated in systemd, systemd will still give it _precedence_ over `unitConfig.StartLimitIntervalSec`.

I noticed that because my `nginx` ignored my `serviceConfig.StartLimitIntervalSec = 0;`, as the nginx NixOS module defines:

https://github.com/NixOS/nixpkgs/blob/5bca1a7664f1769678b2dd8ad0976581a86cda0a/nixos/modules/services/web-servers/nginx/default.nix#L704

(Which, in my opinion, is also a bad default, because I think that web servers should by default retry forever instead of giving up, as individual machines can always have temporary network failures, or talk to machines or infrastructure that have temporary failures, for more than 1 minute.)

**Comment by nh2** [comment]
Subscribing @cyounkins from #45786 so he can see these updates.

**Comment by nh2** [comment]
Probably related PR: #97512

**Comment by nh2** [comment]
Fixed by PR #97512.
